src/Simplex/Chat/View.hs

```diff
@@ -156,62 +156,83 @@ responseToView testView = \case
-          ((msgDirectionInt $ toMsgDirection dir, itemText meta),) $ case quotedItem of
-            Nothing -> Nothing
-            Just CIQuote {chatDir = quoteDir, content} ->
-              Just (msgDirectionInt $ quoteMsgDirection quoteDir, msgContentText content)
+        toChatView :: CChatItem c -> ((Int, Text), Maybe (Int, Text), Maybe String)
+        toChatView (CChatItem dir ChatItem {meta, quotedItem, file}) =
+          ((msgDirectionInt $ toMsgDirection dir, itemText meta), qItem, fPath)
+          where
+            qItem = case quotedItem of
+              Nothing -> Nothing
+              Just CIQuote {chatDir = quoteDir, content} ->
+                Just (msgDirectionInt $ quoteMsgDirection quoteDir, msgContentText content)
+            fPath = case file of
+              Just CIFile {filePath = Just fp} -> Just fp
+              _ -> Nothing
     viewErrorsSummary :: [a] -> StyledString -> [StyledString]
     viewErrorsSummary summary s = [ttyError (T.pack . show $ length summary) <> s <> " (run with -c option to show each error)" | not (null summary)]
 
 viewChatItem :: MsgDirectionI d => ChatInfo c -> ChatItem c d -> [StyledString]
-viewChatItem chat ChatItem {chatDir, meta, content, quotedItem} = case chat of
+viewChatItem chat ChatItem {chatDir, meta, content, quotedItem, file} = case chat of
   DirectChat c -> case chatDir of
     CIDirectSnd -> case content of
-      CISndMsgContent mc -> viewSentMessage to quote mc meta
+      CISndMsgContent mc -> withSndFile to $ sndMsg to quote mc
       CISndDeleted _ -> []
-      CISndFileInvitation fId fPath -> viewSentFileInvitation to fId fPath meta
       where
         to = ttyToContact' c
     CIDirectRcv -> case content of
-      CIRcvMsgContent mc -> viewReceivedMessage from quote meta mc
+      CIRcvMsgContent mc -> withRcvFile from $ rcvMsg from quote mc
       CIRcvDeleted _ -> []
-      CIRcvFileInvitation ft -> viewReceivedFileInvitation from meta ft
       where
         from = ttyFromContact' c
     where
       quote = maybe [] (directQuote chatDir) quotedItem
   GroupChat g -> case chatDir of
     CIGroupSnd -> case content of
-      CISndMsgContent mc -> viewSentMessage to quote mc meta
+      CISndMsgContent mc -> withSndFile to $ sndMsg to quote mc
       CISndDeleted _ -> []
-      CISndFileInvitation fId fPath -> viewSentFileInvitation to fId fPath meta
       where
         to = ttyToGroup g
     CIGroupRcv m -> case content of
-      CIRcvMsgContent mc -> viewReceivedMessage from quote meta mc
+      CIRcvMsgContent mc -> withRcvFile from $ rcvMsg from quote mc
       CIRcvDeleted _ -> []
-      CIRcvFileInvitation ft -> viewReceivedFileInvitation from meta ft
       where
         from = ttyFromGroup' g m
     where
       quote = maybe [] (groupQuote g) quotedItem
   _ -> []
+  where
+    sndMsg to quote mc = do
+      let mcText = msgContentText mc
+      case (mcText, file) of
+        ("", Nothing) -> viewSentMessage to quote mc meta
```

Comment:
in this case nothing will be displayed to user in the terminal view - though it's non trivial to send an empty message currently, if someone manages to do it I think it's better to show that something was received instead of just silently ignoring it

Reply:
this is done by the last case

Reply:
oh.. right.. that's dumb